### ISSUE TITLE
Invalid markup after a table is not detected

### ISSUE THREAD

**Issue by ghost** [sphinx-contrib/sphinx-lint#130]
`sphinx-lint` does not report invalid reST markup after a table, for example:
```
+--------------+----------+-----------+-----------+
| row 1, col 1 | column 2 | column 3  | column 4  |
+--------------+----------+-----------+-----------+
| row 2        |          |           |           |
+--------------+----------+-----------+-----------+

:role:`foo
```

This seems to be caused by the `return` statement in the table detection logic (https://github.com/sphinx-contrib/sphinx-lint/blob/main/sphinxlint/checkers.py#L58).
Changing `return` to `continue` should fix this issue.

**Comment by hugovk** [comment]
Good find, would you like to put together a PR with tests to fix this?

**Comment by koyuki7w** [comment]
@hugovk Thanks, I opened the PR.
